Containerfile

```diff
@@ -1,4 +1,42 @@
+FROM quay.io/centos/centos:stream9 AS builder-octep-cp-agent
```

Comment:
Is this cross compile-able?

Reply:
if [ "$TARGETPLATFORM" = "linux/arm64" ] ; does this mean that only when you are building the container on ARM64 that it would build?

Reply:
the actual building of the agent application only works on arm64. And it's also only useful to have the binary in the arm64 container.

I didn't nicely figure out how to do arch-specific pieces in a multistage Containerfile. With `if [ "$TARGETPLATFORM" = "linux/arm64" ]`, this seems to work as you'd expect.

basically, also on amd64 we will create a `builder-octep-cp-agent` image, but all steps basically do nothing...

Reply:
builder-octep-cp-agent image?

Reply:
the `Containerfile` no is a multi-stage build. The first container is `FROM ... AS builder-octep-cp-agent`. That one builds the octep_cp_agent program, which only actually builds on arm64. But during the mulit-arch build, we create a builder-octep-cp-agent image also for amd64... except, on amd64 it is basically empty due to `if [ "$TARGETPLATFORM" = "linux/arm64" ]`.

Note sure how to make that nicer -- eg. to only have the `builder-octep-cp-agent` image on arm64. For example, Dockerfile don't support an `IF` block to comment out pieces.

But this approach seems to work, so it's probably fine unless proven otherwise.

Reply:
Hmm no problem. Works for me.